src/libraries/System.Text.Json/tests/Serialization/ConstructorTests/ConstructorTests.Cache.cs

```diff
@@ -21,35 +22,35 @@ public void MultipleThreadsLooping()
             }
         }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
         public void MultipleThreads()
         {
             // Verify the test class has >32 properties since that is a threshold for using the fallback dictionary.
             Assert.True(typeof(ClassWithConstructor_SimpleAndComplexParameters).GetProperties(BindingFlags.Instance | BindingFlags.Public).Length > 32);
 
-            void DeserializeObject(string json, Type type, JsonSerializerOptions options)
+            async Task DeserializeObject(string json, Type type, JsonSerializerOptions options)
```

Comment:
nit, for this and subsequent local functions:
```suggestion
            async Task DeserializeObjectAsync(string json, Type type, JsonSerializerOptions options)
```